### PR TITLE
Add a doc on HA

### PR DESCRIPTION
**Pull request by vendrov**
Add a doc on HA

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1239)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
Thanks ... need a couple of minutes to review.  May not get to it till after the holidays.  Hope to see you on office hours tomorrow!

**Comment by vendrov** [comment]
Are you sure it's not on next week?

**Comment by chrislovecnm** [comment]
Yep tomorrow

File: HA.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+## Kops HA
+We can create HA cluster using kops only in the first creation of the cluster by using the "--master-zones" flag
+example: https://github.com/kubernetes/kops/blob/master/docs/advanced_create.md
+K8S relay on a key value DB named "etcd", which is using the the Qurom concept that the cluster need at least 51% of nodes to be
+available, for the cluster to work.
+Kops currently doesn't support cross region HA
+
+As a result there are few considoration that need to take into account when using kops with HA:
```
Comment (apenney) [inline HA.md]: Typo: "considerations that need to be taken into account"

File: HA.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+* Only odd number of masters instances can be created.
```
Comment (apenney) [inline HA.md]: It would be nice to add a reason why!

Comment (apenney) [inline HA.md]: Actually, you explain why just below, maybe just move that up a little :)

Comment (vendrov) [inline HA.md]: which part should move up?

File: HA.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# High Availability (HA)
+## Introduction
+When we create kubernetes cluster using kops we are able to create a multiple K8S masters and by doing so we can assure that in a case of
```
Comment (chrislovecnm) [inline HA.md]: > When we create kubernetes cluster using kops we are able to create a multiple K8S masters and by doing so we can assure that in a case of a master instance failure our K8S functionality won't damaged.

When we create kubernetes cluster using kops we are able to create multiple K8S masters and by doing so we can provide a HA control plane.

^ something like that??

File: HA.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# High Availability (HA)
+## Introduction
+When we create kubernetes cluster using kops we are able to create a multiple K8S masters and by doing so we can assure that in a case of 
+a master instance failure our K8S functionality won't damaged. 
+In scenario of a single master instance that fails our minions/pods will keep running but there will be no new pods schedualing, kubectl
+won't work (since the api-server will be down).
```
Comment (chrislovecnm) [inline HA.md]: We cannot deploy, if a node fails, failover will not be handled properly, and scheduled jobs will not be run :)

File: HA.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+
+## Kops HA
+We can create HA cluster using kops only in the first creation of the cluster by using the "--master-zones" flag
+example: https://github.com/kubernetes/kops/blob/master/docs/advanced_create.md
+K8S relay on a key value DB named "etcd", which is using the the Qurom concept that the cluster need at least 51% of nodes to be
```
Comment (chrislovecnm) [inline HA.md]: > K8S relay on a key value DB named "etcd", which is using the the Qurom concept that the cluster need at least 51% of nodes to be

How about:

K8S relies on a key-value DB named etcd, which is user a Quorum clustering concept.  Three masters are created, each with an etcd database instance.

File: HA.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+available, for the cluster to work.
+Kops currently doesn't support cross region HA
```
Comment (chrislovecnm) [inline HA.md]: Kubernetes does not support this :)

File: HA.md (hunk 5 above)
Comment (chrislovecnm) [inline HA.md]: Here is some more data:

Etcd uses the Raft consensus algorithm to maintain consistency. The algorithm requires that a quorum of nodes be in-sync to make a decision. If you have an n-node cluster, you’ll need n / 2 + 1 nodes for a quorum.

If you want to rewrite that so we are not plagiarizing :)

File: HA.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+As a result there are few considerations that need to be taken into account when using kops with HA:
+* Only odd number of masters instances can be created.
```
Comment (chrislovecnm) [inline HA.md]: n / 2 + 1

File: HA.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+## Kops HA
+We can create HA cluster using kops only in the first creation of the cluster by using the "--master-zones" flag
+example: https://github.com/kubernetes/kops/blob/master/docs/advanced_create.md
+K8S relay on a key value DB named "etcd", which is using the the Qurom concept that the cluster need at least 51% of nodes to be
+available, for the cluster to work.
+Kops currently doesn't support cross region HA
+
+As a result there are few considerations that need to be taken into account when using kops with HA:
+* Only odd number of masters instances can be created. 
+* Currently Kops can't create more than 1 master in a single AZ.
```
Comment (chrislovecnm) [inline HA.md]: Mention that we have an issue open to change this :)

File: HA.md (hunk 7 above)
Comment (chrislovecnm) [inline HA.md]: Should we mention that we create 3 masters in three different AZ?

**Pull request merged**